presto-main/src/main/java/com/facebook/presto/sql/analyzer/MaterializedViewInformationExtractor.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.analyzer;
+
+import com.facebook.presto.sql.tree.ComparisonExpression;
+import com.facebook.presto.sql.tree.DefaultTraversalVisitor;
+import com.facebook.presto.sql.tree.Expression;
+import com.facebook.presto.sql.tree.GroupBy;
+import com.facebook.presto.sql.tree.GroupingElement;
+import com.facebook.presto.sql.tree.Identifier;
+import com.facebook.presto.sql.tree.LogicalBinaryExpression;
+import com.facebook.presto.sql.tree.QuerySpecification;
+import com.facebook.presto.sql.tree.Relation;
+import com.facebook.presto.sql.tree.Select;
+import com.facebook.presto.sql.tree.SingleColumn;
+import com.facebook.presto.sql.tree.Table;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.facebook.presto.sql.analyzer.SemanticErrorCode.NOT_SUPPORTED;
+
+public class MaterializedViewInformationExtractor
+        extends DefaultTraversalVisitor<Void, Void>
+{
+    private MaterializedViewInfo materializedViewInfo = new MaterializedViewInfo();
+
+    @Override
+    protected Void visitQuerySpecification(QuerySpecification node, Void context)
+    {
+        if (node.getLimit().isPresent()) {
+            throw new SemanticException(NOT_SUPPORTED, node, "Limit clause is not supported in query optimizer");
+        }
+        super.visitQuerySpecification(node, context);
+        return null;
+    }
+
+    protected Void visitSelect(Select node, Void context)
+    {
+        super.visitSelect(node, context);
+        materializedViewInfo.setDistinct(node.isDistinct());
+        return null;
+    }
+
+    @Override
+    protected Void visitRelation(Relation node, Void context)
+    {
+        if (node instanceof Table) {
+            materializedViewInfo.setBaseTable(Optional.of(node));
```

Comment:
Lets add another check here. If a base table is already added, this should fail. Basically any MV working with two tables(Join/Union) should not pass from here.